services/director-v2/src/simcore_service_director_v2/models/schemas/dynamic_services/scheduler.py

```diff
@@ -84,32 +87,29 @@ def create_as_initially_ok(cls) -> "Status":
         return initial_state
 
 
-class DockerStatus(str, Enum):
-    CREATED = "created"
-    DEAD = "dead"
-    EXITED = "exited"
-    PAUSED = "paused"
-    REMOVING = "removing"
-    RESTARTING = "restarting"
-    RUNNING = "running"
-
-
 class DockerContainerInspect(BaseModel):
-    status: DockerStatus = Field(
-        ...,
-        scription="status of the underlying container",
+    container_state: ContainerState = Field(
+        ..., description="current state of container"
     )
     name: str = Field(..., description="docker name of the container")
     id: str = Field(..., description="docker id of the container")
 
+    @cached_property
+    def status(self) -> DockerStatus:
+        assert self.container_state.Status  # nosec
+        return self.container_state.Status
+
     @classmethod
     def from_container(cls, container: dict[str, Any]) -> "DockerContainerInspect":
         return cls(
-            status=DockerStatus(container["State"]["Status"]),
+            container_state=ContainerState(**container["State"]),
             name=container["Name"],
             id=container["Id"],
         )
 
+    class Config:
+        keep_untouched = (cached_property,)
```

Comment:
MINOR: perhaps add also that is ``allow_mutation=False`` (i.e. once constructed it cannot be changed) ... otherwise the cache in ``status`` is incorrect.